src/decirest_doc_resource.erl

```diff
@@ -0,0 +1,161 @@
+%%%-------------------------------------------------------------------
+%%% @author mikael
+%%% @copyright (C) 2020, SIGICOM
+%%% @doc
+%%%
+%%% @end
+%%% Created : 23. Nov 2020 22:57
+%%%-------------------------------------------------------------------
+-module(decirest_doc_resource).
+-author("jso").
+
+%% API
+-export([
+  name/0,
+  paths/0,
+  child_of/0,
+  resource_exists/2,
+  fetch_data/2,
+  to_fun/2,
+  to_html/2,
+  doc_map_to_md/1
+]).
+
+name() -> <<"documentation">>.
+
+paths() -> [
+  {"/apidoc/path/[...]", decirest_single_handler},
+  {"/apidoc/name/:name", decirest_single_handler}
+].
+
+child_of() -> [].
+
+resource_exists(Req = #{path := << "/apidoc/path", Path/binary>>}, State = #{ref := Ref, mro_call := true}) ->
+  PathDoc = decirest_doc_lib:fetch_doc_by_path(Ref, Req#{path => Path}, State),
+  parse_doc_output(PathDoc, Req, State);
+resource_exists(Req, State = #{ref := Ref, mro_call := true}) ->
+  %% fetch by name
+  Name = cowboy_req:binding(name, Req),
+  Module = get_module_from_doc_name(Ref, Name),
+  ModuleDoc = decirest_doc_lib:fetch_doc_by_module(Ref, Module, Req, State),
+  parse_doc_output(ModuleDoc, Req, State);
+resource_exists(Req, State) ->
+  {run_default, [Req, State]}.
+
+fetch_data(_Filter, #{docs := Docs}) ->
+  {ok, [Docs]}.
+
+to_fun(Req, State = #{module := Module, rstate := RState}) ->
+  #{Module := #{data := Data}} = RState,
+  Body = doc_map_to_md(Data),
+  {Body, Req, State}.
+
+to_html(Req, State) ->
+  {Body, Req, _State} = to_fun(Req, State),
+  {ok, Css} = get_css_file(),
+  PreMd = <<"<!doctype html>  <html>  <head>", "<meta charset='utf-8'/>  <title>INFRAnet API</title> <style>", Css/binary,
```

Comment:
This line might not be general enough :)

Reply:
Ah I only searched for inapi.